File: src/main/java/org/checkerframework/checker/mungo/typestate/ast/TMethodNode.java
package org.checkerframework.checker.mungo.typestate.ast;

import java.util.List;

public class TMethodNode extends TNode {

  public final String returnType;
  public final String name;
  public final List<String> args;
  public final TNode /*TIdNode | TStateNode | TDecisionStateNode*/ destination;

  public TMethodNode(Position pos, String returnType, String name, List<String> args, TNode destination) {
    super(pos);
    this.returnType = returnType;
    this.name = name;
    this.args = args;
    this.destination = destination;
  }

}

File: src/main/kotlin/org/checkerframework/checker/mungo/typestate/ast/TMethodNode.kt
package org.checkerframework.checker.mungo.typestate.ast

class TMethodNode(pos: Position, val returnType: String, val name: String, val args: List<String>, /*TIdNode | TStateNode | TDecisionStateNode*/val destination: TNode) : TNode(pos) {

  override fun toString(): String {
    return "TMethodNode{${format()}}"
  }

  fun format() = "$returnType $name(${args.joinToString(", ")})"

}
